### PR TITLE
ECDC-3839 CBM data generator

### PR DESCRIPTION
**Pull request by tbukovics**
### Issue reference / description

Intermediate PR to add the Basic generator functionality with TTL and IBM type of beam monitors. Also a generator base time high and low renamed to pulse time high and low.

## Checklist for submitter

- [x] Check for conflict with integration test
- [x] Unit tests pass

---

### Nominate for Group Code Review

- [x] Nominate for code review

### CODE REVIEW

File: src/modules/cbm/CbmInstrument.cpp [hunk 1]
```diff
@@ -7,22 +7,22 @@
...
 #include <common/debug/Trace.h>
 #include <common/readout/ess/Parser.h>
 #include <common/time/TimeString.h>
-#include <cbm/CbmInstrument.h>
-#include <cbm/geometry/Parser.h>
 
-// #undef TRC_LEVEL
-// #define TRC_LEVEL TRC_L_DEB
+#undef TRC_LEVEL
+#define TRC_LEVEL TRC_L_DEB
```
Comment (mortenjc) [inline src/modules/cbm/CbmInstrument.cpp:20]: Forgot to remove debug?

Comment (mortenjc) [reply src/modules/cbm/CbmInstrument.cpp:20]: :-)

File: src/modules/cbm/CbmTypes.h [hunk 2]
```diff
@@ -3,33 +3,90 @@
...
+///     std::cout << "CBM types are different." << std::endl;
+/// }
+/// \endcode
+
+/// \see CbmTypes
+/**
+ * @file CbmTypes.h
+ * @brief Defines the CbmType class and its associated enumeration CbmTypes.
+ */
+
```
Comment (mortenjc) [inline src/modules/cbm/CbmTypes.h:52]: Just noticed this: We generally use the C++ variant for Doxygen comments. I did a quick search and it seems like the /** is something you have introduced

Please for consistency use /// instead
see for example src/common/kafka/AR51Serializer.h

File: src/modules/cbm/CbmTypes.h [hunk 3]
```diff
@@ -62,25 +124,73 @@ class CbmType {
...
-    operator int() const {
-        return static_cast<int>(beamMonitorType);
-    }
+  /**
+   * @brief Conversion operator to int.
+   * @return The integer representation of the CBM type.
+   */
+  operator int() const { return static_cast<int>(beamMonitorType); }
 
-    operator CbmTypes() const {
```
Comment (mortenjc) [inline src/modules/cbm/CbmTypes.h:77]: I don't like operator overload for (). IMO it is fine to do it for == for which the intention is clear
but () is confusing.

We have had a really good example for a bad use of too much operator overloading in the past (JADAQ repository for the old multiblade readout system). Please reconsider this.

**Review by mortenjc — CHANGES_REQUESTED** [review]

File: src/modules/cbm/CbmInstrument.cpp (hunk 1 above)
Comment (tbukovics) [reply src/modules/cbm/CbmInstrument.cpp:20]: :D

File: src/modules/cbm/CbmTypes.h (hunk 2 above)
Comment (tbukovics) [reply src/modules/cbm/CbmTypes.h:52]: Yes, sorry it's comming from my JAVA habbits. I will take care for the future.

**Pull request merged**